selftests/functional/test_list.py

```diff
@@ -0,0 +1,34 @@
+import os
+import unittest
+
+from avocado.core import exit_codes
+from avocado.utils import process
+
+from .. import AVOCADO, BASEDIR
+
+
+class List(unittest.TestCase):
+
+    list_command = 'list'
+    instrumented = 'INSTRUMENTED'
+
+    def test_list_filter_by_tags(self):
+        examples_dir = os.path.join(BASEDIR, 'examples', 'tests')
+        cmd_line = "%s --verbose %s -t fast -- %s" % (AVOCADO,
+                                                      self.list_command,
+                                                      examples_dir)
+        result = process.run(cmd_line)
+        self.assertEqual(result.exit_status, exit_codes.AVOCADO_ALL_OK,
+                         "Avocado did not return rc %d:\n%s"
+                         % (exit_codes.AVOCADO_ALL_OK, result))
+        stdout_lines = result.stdout_text.splitlines()
+        self.assertIn("TEST TYPES SUMMARY", stdout_lines)
+        self.assertIn("%s: 2" % self.instrumented, stdout_lines)
+        self.assertIn("TEST TAGS SUMMARY", stdout_lines)
+        self.assertIn("fast: 2", stdout_lines)
```

Comment:
Every time we add a test with tag `fast`, this test will fail, but it will not be a regression.

One possibility to overcome this would be creating a directory with some dummy tests with the `fast` tag, for example, 2 or 3, then run the `list` on this directory. On a fail, we know it is a possible regression and not a parameter change.

This approach is a bit more complicated and time-consuming to implement, but I think it is more reliable. Anyway, the current code works and will need some debugging when the parameter (number of tags found) changes, so it is just a note, not even a suggestion :)

Reply:
> Every time we add a test with tag `fast`, this test will fail, but it will not be a regression.
> 
> One possibility to overcome this would be creating a directory with some dummy tests with the `fast` tag, for example, 2 or 3, then run the `list` on this directory. On a fail, we know it is a possible regression and not a parameter change.
> 

Yeah, I agree that it has its downsides.  On the other hand, I don't think we touch those tags so often that it justifies all the steps (and maintenance) associated with creating those dummy tests.

> This approach is a bit more complicated and time-consuming to implement, but I think it is more reliable. Anyway, the current code works and will need some debugging when the parameter (number of tags found) changes, so it is just a note, not even a suggestion :)

Noted... and we should catch the failure within the same commit/PR, which should make it pretty clear what's going on.